### PR TITLE
Implement eth_getBalance

### LINKED ISSUE #765: ETH RPC balance API
Return GasCoin balance

### PR DESCRIPTION
**Pull request by YusongWang**
## Summary

- Closes #765

### CODE REVIEW

**Comment by YusongWang** [comment]
@jolestar I need some tips

- How to initialize the account with Genesis or other ways? 
- Also register eth address for the publickey or rooch address?

Thanks.

Also, does someone else know this?  PLS give me a brife

**Comment by jolestar** [comment]
> @jolestar I need some tips
> 
> * How to initialize the account with Genesis or other ways?
> * Also register eth address for the publickey or rooch address?
> 
> Thanks.
> 
> Also, does someone else know this? PLS give me a brief

Send a transaction via MetaMask and ETH RPC will auto-create an account and bind the EHTAddress. After the #735 is finished, the developer can send ETH transactions via CLI.

**Comment by jolestar** [comment]
Record a issue about the MetaMask document #773

**Comment by jolestar** [comment]
Hi, @YusongWang. Do you want to continue this PR?

**Comment by YusongWang** [comment]
> Hi, @YusongWang. Do you want to continue this PR?

Yeah, but still working on fill up gas coin, not really working on the query part!

**Comment by YusongWang** [comment]
Hi @jolestar @baichuan3 @wow-sven @templexxx   I post new work here.

the issue I have for now is when I send a tx in eth.

the contract call will fail and give a code I can't find where is and can't pass by any mock code.

```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: status ABORTED of type Execution with sub status 66538', crates/rooch-rpc-server/src/service/mod.rs:58:14
```

Can you look at this?

File: crates/rooch-types/src/transaction/ethereum.rs [hunk 1]
```diff
@@ -26,11 +26,51 @@ use serde::{Deserialize, Serialize};
...
+            let serialized = test.to_bytes();
+
+            // Handle the transfer gas coin function call only in ethereum
+            let payload = MoveAction::new_function_call(
+                moveos_types::move_types::FunctionId::new(
+                    move_core_types::language_storage::ModuleId::new(
+                        AccountAddress::from_hex_literal("0x3").unwrap(),
+                        Identifier::new("coin").unwrap(),
+                    ),
+                    Identifier::new("transfer").unwrap(),
```
Comment (jolestar) [inline crates/rooch-types/src/transaction/ethereum.rs:44]: The GasCoin currently does not support public transfer. If we want to support transfer coin, we need to add the `store` ability to the `GasCoin` struct. Or provide a `gas_coin::transfer` function.

**Comment by wow-sven** [comment]
> Hi @jolestar @baichuan3 @wow-sven @templexxx I post new work here.
> 
> the issue I have for now is when I send a tx in eth.
> 
> the contract call will fail and give a code I can't find where is and can't pass by any mock code.
> 
> ```
> thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: status ABORTED of type Execution with sub status 66538', crates/rooch-rpc-server/src/service/mod.rs:58:14
> ```
> 
> Can you look at this?

The suggested solution to this issue is to modify the gas of the transaction. By increasing the gas limit, it may provide enough resources for the transaction to complete successfully. You can find more details and discussion about this issue in the following link: [rooch-network/rooch issue #842.](https://github.com/rooch-network/rooch/issues/842)


refer to https://github.com/rooch-network/rooch/blob/main/moveos/moveos-types/src/gas_config.rs

**Comment by YusongWang** [comment]
@jolestar I quit, I can't make any progress. sorry for that, and same to me.

**Comment by jolestar** [comment]
> @jolestar I quit, I can't make any progress. sorry for that, and same to me.
> 
> 
> 
> 

It is ok, the system is complex and the documentation is not complete. Thank you for contributions.

**Pull request closed without merging**